File: riveredge-backend/src/core/schemas/audit_binding.py
"""审核单据绑定 Schema"""

from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, Field, ConfigDict


class AuditBindingItemResponse(BaseModel):
    """单条审核绑定（含 manifest 声明信息）"""

    node_key: str
    entity_type: str
    resource: str
    name: str
    app: str
    config_category: str
    template: str
    is_enabled: bool
    process_uuid: Optional[UUID] = None
    process_name: Optional[str] = None
    process_code: Optional[str] = None


class AuditProcessOptionResponse(BaseModel):
    """可选审批流程（流程库）"""

    uuid: UUID
    name: str
    code: str
    is_active: bool

    model_config = ConfigDict(from_attributes=True)


class AuditBindingListResponse(BaseModel):
    """审核设置列表"""

    items: List[AuditBindingItemResponse]
    process_options: List[AuditProcessOptionResponse]


class AuditBindingUpdateRequest(BaseModel):
    """更新审核绑定"""

    is_enabled: Optional[bool] = Field(None, description="是否启用人工审核")
    process_uuid: Optional[UUID] = Field(None, description="绑定的审批流程 UUID")

File: riveredge-frontend/src/services/auditBinding.ts
/**
 * 审核单据绑定（配置中心审核设置）
 */

import { apiRequest } from './api';

export interface AuditBindingItem {
  node_key: string;
  entity_type: string;
  resource: string;
  name: string;
  app: string;
  config_category: string;
  template: string;
  is_enabled: boolean;
  process_uuid?: string | null;
  process_name?: string | null;
  process_code?: string | null;
  process_matched?: boolean;
}

export interface AuditProcessOption {
  uuid: string;
  name: string;
  code: string;
  is_active: boolean;
}

export interface AuditBindingListResponse {
  items: AuditBindingItem[];
  process_options: AuditProcessOption[];
}

export interface AuditBindingUpdatePayload {
  is_enabled?: boolean;
  process_uuid?: string | null;
}

export async function getAuditBindings(): Promise<AuditBindingListResponse> {
  return apiRequest<AuditBindingListResponse>('/core/audit-bindings');
}

export async function updateAuditBinding(
  nodeKey: string,
  data: AuditBindingUpdatePayload,
): Promise<{ node_key: string; is_enabled: boolean; process_uuid?: string | null }> {
  return apiRequest(`/core/audit-bindings/${encodeURIComponent(nodeKey)}`, {
    method: 'PUT',
    data,
  });
}
